src/Aspire.Hosting/Dashboard/CommandsConfigurationExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal static class CommandsConfigurationExtensions
+{
+    internal static IResourceBuilder<T> WithLifeCycleCommands<T>(this IResourceBuilder<T> builder) where T : IResource
+    {
+        builder.WithCommand(
+            "start",
+            "Start",
+            resourceSnapshot => Task.FromResult(IsStopped(resourceSnapshot.State?.Text)),
```

Comment:
This should probably be delegated to ApplicationExecutor as well, because the logic that determines whether something can be started/stopped/restarted may be more complicated than f: resourceState -> bool.

Reply:
The resource snapshot data comes from ApplicationExecutor. If we need more information on it to make this decision I think it would be better to have it available on the model, and then use it here.

Reply:
OK that makes sense